dht22_exporter.py

```diff
@@ -41,11 +39,21 @@ def read_sensor():
 
 
 def main():
-    start_http_server(8001)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-g", "--gpio", dest="gpio", type=int, 
+                        required=True, help="GPIO pin number on which the sensor is plugged in")
+    parser.add_argument("-i", "--interval", dest="interval", type=int, 
+                        required=True, help="Interval sampling time, in seconds")
+    parser.add_argument("-p", "--port", dest="port", type=int, default=8001, 
+                        required=False, help="Port that will be exposed")
+
+    args = parser.parse_args()
+
+    start_http_server(args.port)
 
     while True:
-        read_sensor()
-        time.sleep(int(sys.argv[2]))
+        read_sensor(pin=args.gpio)
+        time.sleep(int(args.interval))
```

Comment:
Does this need to be cast to an int if we already specify it's an int via argparse?

Reply:
you're definitely right, there is no need for the integer casting in the sleep